jumanji/environments/combinatorial/routing/generation_methods_v1_uo.py

```diff
@@ -0,0 +1,572 @@
+import random
+from collections import deque
+from typing import List, Tuple, Union, Optional, Dict, Callable
+import numpy as np
+
+
+class Grid:
+    def __init__(self, rows: int, cols: int, grid: Optional[np.ndarray] = None) -> None:
+        """
+        Constructor for the Grid class.
+        Args:
+            rows (int): number of rows in the grid
+            cols (int): number of columns in the grid
+            grid (np.array): grid layout
+        Returns:
+            None
```

Comment:
Check google style guide but I think if the function doesnt return anything you dont have to write a returns

Reply:
Also check about how to write the init - I think you dont have to say it's a constructor